witness/src/Dscp/Snowdrop/AccountValidation.hs

```diff
@@ -112,89 +119,58 @@ preValidateSimpleMoneyTransfer =
         -- Strip Update ctor from values.
         changes <- mapM checkThatAccountIsUpdatedOnly accountChanges
 
-        let (payer, recipients) = List.partition (is author) changes
+        let isAuthor :: (AccountId, Account) -> Bool
+            isAuthor (accId, _) = accId == author
+        let (payer, recipients) = List.partition isAuthor changes
+
+        allOutputs <- mapM validateSaneArrival recipients
 
-        (paid, deltas) <-
-            (,) <$> validateSaneDeparture payer before
-                <*> mapM validateSaneArrival recipients
+        let balanceBefore = aBalance before
+        -- Total amount of coins sent (sum of outputs)
+        let receivedTotal = sum allOutputs
+        -- Previous value minus fees
+        let receivedNoFees = receivedTotal - coinToInteger (unFees fees)
```

Comment:
I'm stuck here (
What physical sense `sum outs - fee` has?

Reply:
No, that's not a bug. Expander adds fees sent as a separate output, so in the validator we have an extra output, which is counted in `receivedTotal`. I've added a comment about that.

Reply:
Okay, it's still not clear from the comment that fees are _already_ included in the outputs after expansion, but fine.

Reply:
It is not clear from the comment, but the careful_programmer^tm should also check the expander, because some checks and predicates are located there (because expander/verifier framework is not ideal). So when he reads the expander he'll get why it is made so.
)